.pre-commit-config.yaml

```diff
@@ -42,6 +43,15 @@ repos:
     rev: v1.4.0
     hooks:
     -   id: rst-backticks
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.711
+    hooks:
+    -   id: mypy
+        name: mypy (src)
+        files: ^src/
+    -   id: mypy
+        name: mypy (testing)
+        files: ^testing/
```

Comment:
oh, I meant you'll probably want one of these for the other dirs you had in the original patch (`extras`, `bench`, etc.) -- it's probably ok to not check testing (I was just using that as an example)

Reply:
I don't think it's worth checking them TBH - they're just some scripts that are probably easier to handle without mypy over your shoulder.

For testing/ I think it's valuable to have type checking, but can be removed later if it's too much to handle.

Reply:
the strategy I usually use for testing is `check_untyped_defs` and don't require annotations -- it strikes a nice balance between checled and tedium (generally there's no reason to type check tests because they always run anyway -- and they often run into *intentional* type issues when checking how an api interacts with incorrect types) :man_shrugging:

Reply:
Makes sense. For now, the level of type checking is about what you describe, for the entire project (although `check_untyped_defs` is not activated yet - probably the next logical step).

Reply:
sounds good -- thanks for putting so much work into this, this is great :tada: